pkg/utils/node.go

```diff
@@ -15,6 +15,291 @@ import (
 	netutils "k8s.io/utils/net"
 )
 
+// nodeAddressMap is a mapping of address types to a list of addresses of that type.
+// It preallocates the slices of addresses.
+type nodeAddressMap map[apiv1.NodeAddressType][]apiv1.NodeAddress
+
+// add adds an address of the given type to the address map. If the given type
+// was not already in the map, it creates a new preallocated entry for it.
+func (m nodeAddressMap) add(address apiv1.NodeAddress) {
+	if _, ok := m[address.Type]; ok {
+		m[address.Type] = append(m[address.Type], address)
+	} else {
+		m[address.Type] = make([]apiv1.NodeAddress, 0)
+		m[address.Type] = append(m[address.Type], address)
+	}
+}
+
+// nodeAddressMap is a mapping of address types to a list of addresses of that type.
+// It preallocates the slices of addresses.
+type addressMap map[apiv1.NodeAddressType][]net.IP
+
+// KRNode is a struct that holds information about a node that is used by kube-router.
+type KRNode struct {
+	NodeInterfaceName string
+	NodeIPv4Addrs     addressMap
+	NodeIPv6Addrs     addressMap
+	NodeName          string
+	linkQ             LocalLinkQuerier
+	PrimaryNodeSubnet net.IPNet
```

Comment:
I think I prematurely added this to the interface because of the usage of `utils.GetNodeSubnet(nodeIP, nil)` in `network_routes_controller.go`.

You're right, it isn't used anywhere, I'll remove it.